From 9d20f92aa78878b75ea725ee64ff84e4f42af106 Mon Sep 17 00:00:00 2001
From: Subrata Chatterjee <schatterjee@habana.ai>
Date: Tue, 13 Jun 2023 08:10:52 +0300
Subject: [PATCH] [SW-148105] gaudi3: sec: open rr config if non-secured

So far in ppboot (gaudi3) RR protections for lbw,
hbw config space are opened for PLDM target.

This patch modifies the implementation to open only if
security is not enabled in efuse.
Same is done for both pldm and asic targets.

Change-Id: Ia559377de336750c770dad5c0039ab8d86f346dc
Signed-off-by: Subrata Chatterjee <schatterjee@habana.ai>
---

diff --git a/arch/gaudi3/arc/asic.c b/arch/gaudi3/arc/asic.c
index 4b9f239..44ca56d 100644
--- a/arch/gaudi3/arc/asic.c
+++ b/arch/gaudi3/arc/asic.c
@@ -101,3 +101,9 @@
 
 	hl_pif_clear_sts(PIF_STAT_POLL_PCIE_PHY_LOAD);
 }
+
+void ppboot_security_init(void)
+{
+	hl_dev_security_init();
+}
+
diff --git a/arch/gaudi3/arc/common.c b/arch/gaudi3/arc/common.c
index ab16f65..acf5116 100644
--- a/arch/gaudi3/arc/common.c
+++ b/arch/gaudi3/arc/common.c
@@ -1341,3 +1341,19 @@
 	/* D0 PIF ARC, has LBW bus only*/
 	hl_set_mstr_if_axprot_to_priv(mmD0_PIF_ARC_MSTR_IF_AXPROT_LBW_BASE);
 }
+
+void hl_dev_security_init(void)
+{
+	u32 val;
+	int err;
+
+	err = preboot_fuse_read(EFUSE_SYSCFG_OFFSET, &val);
+	if (err)
+		return;
+
+	/* TODO: Do meaningful security config incase SEC_EN is set */
+	if (val & EFUSE_SEC_EN_MASK)
+		return;
+
+	security_rr_open_pcie_config();
+}
diff --git a/arch/gaudi3/arc/pldm.c b/arch/gaudi3/arc/pldm.c
index d1ac6a3..3eb3497 100644
--- a/arch/gaudi3/arc/pldm.c
+++ b/arch/gaudi3/arc/pldm.c
@@ -149,7 +149,7 @@
 
 void ppboot_security_init(void)
 {
-	security_rr_open_pcie_config();
+	hl_dev_security_init();
 }
 
 bool hl_dev_is_valid_img_desc(struct boot_desc *desc)
diff --git a/include/preboot.h b/include/preboot.h
index c57956c..741ee11 100644
--- a/include/preboot.h
+++ b/include/preboot.h
@@ -264,6 +264,7 @@
 void ppboot_check_stack_mon_dev_specific(void);
 void ppboot_pcie_drain_init(void);
 void ppboot_security_init(void);
+void hl_dev_security_init(void);
 int ppboot_boot_info_get(u32 type, void *data, u32 max_size, u32 *copy_size);
 void ppboot_plls_init(void);
 
